timeseries/src/autogluon/timeseries/metrics/abstract.py

```diff
@@ -0,0 +1,155 @@
+from typing import List, Optional
+
+import numpy as np
+import pandas as pd
+
+from autogluon.timeseries import TimeSeriesDataFrame
+from autogluon.timeseries.utils.datetime import get_seasonality
+from autogluon.timeseries.utils.warning_filters import warning_filter
+
+
+class TimeSeriesScorer:
+    """Base class for all evaluation metrics used in AutoGluon-TimeSeries.
+
+    This object always returns the metric in greater-is-better format.
+
+    Follows the design of ``autogluon.core.metrics.Scorer``.
+
+    Attributes
+    ----------
+    greater_is_better : bool, default = False
+        Whether internal method :meth:`~autogluon.timeseries.metrics.TimeSeriesScorer.compute_metric` is
+        a loss function (default), meaning low is good, or a score function, meaning high is good.
+    optimum : float, default = 0.0
+        The best score achievable by the score function, i.e. maximum in case of scorer function and minimum in case of
+        loss function.
+    optimized_by_median : bool, default = False
+        Whether given point forecast metric is optimized by the median (if True) or expected value (if False). If True,
+        all models in AutoGluon-TimeSeries will attempt to paste median forecast into the "mean" column.
+    """
+
+    greater_is_better: bool = False
+    optimum: float = 0.0
+    optimized_by_median: bool = False
```

Comment:
Existing problem: Depending on the metric specified by the user, the optimal point forecast can be either in the `"mean"` column (e.g., RMSE) or in the `0.5` quantile column (e.g., `MAE`, `MASE`). This is a high burden to place on the user.

Our solution for v1.0: We always return the forecast that optimizes the chosen `eval_metric` under the `"mean"` column. This means, all models need to know what to put in the `"mean"` column (conditional mean or median?). We convey this information using the `optimized_by_median` attribute of the metric.